### PR TITLE
Introducing "locked owner su vault" structs-only in validator + Renames

### PR DESCRIPTION
**Pull request by jakrawcz-rdx**
⚠️ This PR builds on top of https://github.com/radixdlt/radixdlt-scrypto/pull/1029

This PR **only** introduces structs needed for **Q5-B** of https://radixdlt.atlassian.net/wiki/spaces/S/pages/3058827265/REP-83+Calculating+Emissions+and+Handling+Validator+Fees
(the impl will come in some next PR)

Also, this PR applies a few renames (mostly regarding consistent naming of "stake unit" inside validator) and adds missing rustdocs _(mostly for my own convenience, since I am recently reading a lot of code there and I sometimes forget whether some amount is XRD or SU...)_.
So, **let's use it to establish some naming** (and avoid discussing names in the coming more logic-heavy PRs).

### CODE REVIEW

**Review by dhedey — APPROVED** [review]
Minor thoughts on naming 👍

File: radix-engine-interface/src/blueprints/epoch_manager/invocations.rs [hunk 1]
```diff
@@ -209,7 +210,7 @@ pub const VALIDATOR_UNSTAKE_IDENT: &str = "unstake";
 
 #[derive(Debug, Eq, PartialEq, ScryptoSbor)]
 pub struct ValidatorUnstakeInput {
-    pub lp_tokens: Bucket,
+    pub su_bucket: Bucket,
```
[the remark this answers is not in the document]
Comment (jakrawcz-rdx) [reply radix-engine-interface/src/blueprints/epoch_manager/invocations.rs:213]: Let me go for `stake_unit` consistently then 👍 
(still, there will be usages like `stake_unit_token` to represent the "resource" of the stake unit; `stake_unit_vault`, `stake_unit_bucket`, etc. too)

File: radix-engine-tests/tests/epoch_manager.rs [hunk 2]
```diff
@@ -561,14 +561,14 @@ fn validator_set_receives_emissions_proportional_to_stake_on_epoch_change() {
         vec![
             RewardAppliedEvent {
                 stake_added_xrd: a_stake_added,
-                liquidity_token_supply: a_stake,
+                total_su_supply: a_stake,
```
[the remark this answers is not in the document]
Comment (jakrawcz-rdx) [reply radix-engine-tests/tests/epoch_manager.rs:564]: > The total_su_supply supply isn't equal to a_stake in general - or at least it's not after there have been any emissions.

No, not in general, but in my case I have captured the stake before any emissions as `a_stake`, so I can safely assert on this. Let me rename this to `a_initial_stake`?
(TBH I think it is cool to assert that "SU supply after emission stays the same as before emission [which is equal to XRD stake before emission]")

File: radix-engine-interface/src/blueprints/epoch_manager/invocations.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (dhedey) [reply radix-engine-interface/src/blueprints/epoch_manager/invocations.rs:213]: :+1: sounds good.

(Although on that note, I feel `stake_unit_resource` or `stake_unit_address` is possibly better than `stake_unit_token` - we've generally avoided using the term "token" inside the engine)

**Pull request merged**